docs/docs/building-blocks/6-optimizers.md

```diff
@@ -38,36 +38,32 @@ All of these can be accessed via `from dspy.teleprompt import *`.
+3. **`BootstrapFewShotWithRandomSearch`**: Applies `BootstrapFewShot` several times with random search over generated demonstrations, and selects the best program over the optimization. Parameters mirror those of `BootstrapFewShot`, with the addition of `num_candidate_programs`, which specifies the number of random programs evaluated over the optimization, including candidates of the uncompiled program, `LabeledFewShot` optimized program, `BootstrapFewShot` compiled program with unshuffled examples and `num_candidate_programs` of `BootstrapFewShot` compiled programs with randomized example sets. 
 
-4. **`BootstrapFewShotWithOptuna`**: Applies `BootstrapFewShot` through Optuna hyperparameter optimization across demonstration sets, running trials to maximize evaluation metrics. As before, chooses the best of a candidate set of demonstration examples, but this time using Optuna to make the choice. <!--- TBQH, I don't understand how Optuna does this. As far as I can tell it simply chooses best based on multiple evaluations, rather than a single one, and mention of "hyperparameters" seems to be a red herring. ---> 
+4. **`BootstrapFewShotWithOptuna`**: Applies `BootstrapFewShot` through Optuna hyperparameter optimization across demonstration sets, running trials to maximize evaluation metrics and selecting the best demonstrations. 
 
-5. **`KNNFewShot`**. Selects demonstrations through k-Nearest Neighbors algorithm.  Vectorizes the examples, and then clusters them, using cluster centers with `BootstrapFewShot` for bootstrapping/selection process. <!-- I'm not at all sure that this is right. I couldn't follow the KNN code in the repo, so I just assumed that dspy was trying to cover the space of possible examples by picking centers of different clusters. -->
+5. **`KNNFewShot`**. Selects demonstrations through k-Nearest Neighbors algorithm.  Vectorizes the examples, and then clusters them, using cluster centers with `BootstrapFewShot` for bootstrapping/selection process. 
 
 
 
 #### Automatic Instruction Optimization
 
-Unlike the above optimizers, these attempt to optimize the instructions, rather than the questions.
+These optimizers serve to produce optimal instructions for the prompt, in addition to optimized few-shot demonstrations.
 
-6. **`COPRO`**: Uses an LLM to generate and refine new instructions for each step, and optimizes them with coordinate ascent (hill-climbing using the metric function and the `trainset`).  The number of iterations of prompt improvement is controlled by the `depth` parameter.
-
-#### Simultaneous optimization
```

Comment:
@arnavsinghvi11 The reason I split this into two sections with the different headers is that it looked like COPRO only optimizes the instructions, and not the example set, whereas MIPRO optimizes both.

If that's the case, then I think my structure is better. Again, I may have misunderstood.

Reply:
I see - I think it's fine to group COPRO and MIPRO in terms of *Signature Optimizers*, with the mentioning that MIPRO also provides optimized few-shot examples. (Having individual sections for each optimizer makes the sections a bit unneeded :).